docs/portal.properties-Reference.md

```diff
@@ -280,3 +281,62 @@ This gene set will add the following in the query box:
 ```
 "BRCA genes" BRCA1: MUT=E1258D; BRCA2: HOMDEL MUT=NONSENSE MUT=NONSTART MUT=NONSTOP MUT=FRAMESHIFT MUT=SPLICE MUT=TRUNC;
 ```
+# Ehcache Settings
+cBioPortal is supported on the backend with Ehcache. The configuration, size, and location of these caches are configurable from within portal.properties through the following properties.
+
+First, select a cache configuration using `ehcache.xml_configuration`. This specifies whether to use a disk-only, heap-only, or hybrid (disk + heap) cache. The default value is `ehcache-mixed.xml` which initializes a hybrid caching system.
+```
+ehcache.xml_configuration=[ehcache-heap-only.xml OR ehcache-mixed.xml OR ehcache-disk-only.xml]
+```
+
+If the cache is configured to use disk resources, users must make a directory available and set it with the `ehcache.persistence_path` property. Ehcache will create seperate directories under the provided path for each cache defined in the ehcache.xml_configuration file. 
+```
+ehcache.persistence_path=[location on the disk filesystem where Ehcache can write the cache to /tmp/]
+```
+
+Cache size must be set for both heap and disk; Ehcache requires disk size to be greater than heap size. Default values are provided. The general repository cache is specified to use 1021MB for heap and 4GB for disk. The static repository cache is specified to use 30MB for heap and 32MB for disk. For installations with increased traffic or data, cache sizes can be increased to further improve performance. 
+```
+ehcache.general_repository_cache.max_bytes_heap=
+ehcache.general_repository_cache.max_bytes_heap_units=[size unit e.g MB, GB, entries]
```

Comment:
Personally, I prefer less property settings to prevent issues. I don't see anyone would want to set this to B, or TB so maybe default to MB? If they want to increase to GB level, just need to do the math?

Reply:
When the user disables Ehcache (ehcache.cache_enabled=false) they still must give a valid configuration, because cache_enabled only means that items are never put in the cache or retrieved from the cache ... not that the cache does not exist. So in the documentation we wrote, we recommend setting a heap-only configuration and setting the size to something minimal (1B). I guess 1MB is not that much space to waste (and maybe Ehcache does not allocate any memory until items are actually stored in the cache). So we could go with MB as the default unit size if we are willing to possibly waste 1 MB of heap space. Anyone have comments?